optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingImporter.java

```diff
@@ -0,0 +1,496 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.conferencescheduling.persistence;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceParametrization;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
+import org.optaplanner.examples.conferencescheduling.domain.Room;
+import org.optaplanner.examples.conferencescheduling.domain.Speaker;
+import org.optaplanner.examples.conferencescheduling.domain.Talk;
+import org.optaplanner.examples.conferencescheduling.domain.TalkType;
+import org.optaplanner.examples.conferencescheduling.domain.Timeslot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConferenceSchedulingImporter {
+
+    private static final Logger logger = LoggerFactory.getLogger(ConferenceSchedulingImporter.class);
+    private static final String zoneId = "Europe/Paris";
+    private static final String[] smallRoomsTypeNames = {"Quickie Sessions", "Quickie", "Hands-on Labs", "BOF (Bird of a Feather)"};
+    private static final String[] mediumRoomsTypeNames = {"Tools-in-Action", "University", "Conference", "Deep Dive",
+            "Opening Keynote", "Closing Keynote", "Keynote", "Ignite Sessions"};
+    private static final String[] largeRoomsTypeNames = {"break"};
+    private static final String SMALL_ROOM_TAG = "small";
+    private static final String MEDIUM_ROOM_TAG = "medium";
+    private static final String LARGE_ROOM_TAG = "large";
+
+    private ConferenceSchedulingRESTEndpoints endpoints;
+    private Map<String, TalkType> talkTypeNameToTalkTypeMap;
+    private Map<String, Room> roomIdToRoomMap;
+    private Map<String, Speaker> speakerNameToSpeakerMap;
+    private Map<String, Talk> talkCodeToTalkMap;
+    private Set<String> talkUrlSet;
+    private Set<String> trackIdSet;
+
+    private ConferenceSolution solution;
+
+    public ConferenceSchedulingImporter() {
+        this.endpoints = new ConferenceSchedulingRESTEndpoints();
+        //TODO: Add a panel to get these from the user
+        this.endpoints.setBaseUrl("https://dvbe18.confinabox.com/api/conferences/DVBE18");
```

Comment:
If you get all the files locally, you should be able to use something like this here for mocking out purposes:
`  setBaseUrl("file:///home/musa/projects/optaplanner/optaplanner-examples/local/cfpDevoxxMock/")`

Reply:
Also, when getting the baseUrl supplied from the user in a dialog, there should be some code to deal with the presence or absence of the trailing "/". Later code presumes it's not there (it probably should presume it's there), so the input needs to automatically add/remove the trailing "/" accordingly.